internal/service/advancedclustertpf/resource.go

```diff
@@ -190,7 +191,12 @@ func (r *rs) createCluster(ctx context.Context, plan *TFModel, diags *diag.Diagn
 	if pauseAfter {
 		latestReq.Paused = nil
 	}
-	_, _, err = api.CreateCluster(ctx, projectID, latestReq).Execute()
+	if usingLegacySchema(ctx, plan.ReplicationSpecs, diags) {
```

Comment:
can we add an acc test that would fail before this change?

Reply:
See comment here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2892/files#r1885034045 
I don't know if we have any test that would have a failure by using this old API as I'm not 100% sure that it triggers ISS or auto_scaling_mode=SHARD 🤔